File: Burgler/Burgler.Entities/Ingredients/Ingredients.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Burgler.Entities.IngredientsNS
{
    public static class Ingredients
    {
        public static Buns Buns => new Buns();
        public static Patties Patties => new Patties();
        public static Toppings Toppings => new Toppings();
    }
}

File: Burgler/BurglerApp/ClientApp/src/app/menu/ingredients.ts
export class Ingredients {
  buns: Bun[] = [];
  patties: Patty[] = [];
  toppings: Topping[] = [];
}

export class Ingredient {
  name: string;
  calories: number;
  price: number;
  description: string;
}

class Bun extends Ingredient {}
class Topping extends Ingredient {}
class Patty extends Ingredient {
  size: string;
}
